tests/test_servicecatalog.py

```diff
@@ -0,0 +1,83 @@
+            session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['Id'], 'port-tb63f2b33cska')
+        if self.recording:
+            time.sleep(10)
+        client = session_factory().client("servicecatalog")
+        portfolios = client.list_portfolios()
+        self.assertFalse('port-tb63f2b33cska' in [p.get(
+            'Id') for p in portfolios.get('PortfolioDetails')])
+
+    def test_portfolio_cross_account_remove(self):
+        session_factory = self.replay_flight_data("test_portfolio_cross_account_remove")
+        client = session_factory().client("servicecatalog")
+        accounts = client.list_portfolio_access(PortfolioId='port-hlgxpz7lc55iw').get('AccountIds')
+        self.assertEqual(len(accounts), 1)
+        p = self.load_policy(
+            {
+                "name": "servicecatalog-portfolio-cross-account",
+                "resource": "catalog-portfolio",
+                "filters": [{"type": "cross-account"}],
+                "actions": [{"type": "remove-shared-accounts", "accounts": "matched"}],
+            },
+            session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['Id'], 'port-hlgxpz7lc55iw')
+        response = client.list_portfolio_access(PortfolioId='port-hlgxpz7lc55iw').get('AccountIds')
+        self.assertEqual(len(response), 0)
+
+    def test_remove_accounts_validation_error(self):
+        self.assertRaises(
+            PolicyValidationError,
+            self.load_policy,
+            {
+                "name": "catalog-portfolio-delete-shared-accounts",
+                "resource": "aws.catalog-portfolio",
+                "actions": [{"type": "remove-shared-accounts", "accounts": "matched"}],
+            }
+        )
+
+    def test_portfolio_remove_share_accountid(self):
+        session_factory = self.replay_flight_data("test_portfolio_remove_share_accountid")
+        client = session_factory().client("servicecatalog")
+        self.assertTrue('644160558196' in client.list_portfolio_access(
+            PortfolioId='port-hlgxpz7lc55iw').get('AccountIds'))
+        self.assertTrue('644160558196' in client.list_portfolio_access(
+            PortfolioId='port-srkytozjwbzpc').get('AccountIds'))
+        p = self.load_policy(
+            {
+                "name": "servicecatalog-portfolio-cross-account",
+                "resource": "catalog-portfolio",
+                "filters": [{"type": "cross-account"}],
+                "actions": [{"type": "remove-shared-accounts", "accounts": ["644160558196"]}],
+            },
+            session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 2)
+        self.assertTrue(r['Id'] in ['port-hlgxpz7lc55iw', 'port-srkytozjwbzpc'] for r in resources)
+        self.assertFalse('644160558196' in client.list_portfolio_access(
```

Comment:
This account number is not the one used for real testing but changed after the tests were executed successfully, since this is the default account we use